### DISCUSSION TITLE
EvaluateTx historical tx

### DISCUSSION THREAD

**Discussion by szist** [CardanoSolutions/ogmios#276] [Ideas / Feature Requests]
Correct me if I'm wrong currently this is how evaluate tx endpoint works:

1. accept a signed Tx to evaluate from the endpoint (and an additional utxo set)
2. collect referenced utxos from the state query client
3. query protocol parameters from the state query client
4. use ledger tx evaluation function without using any mini-protocol or communication with the node

Having 3 might not be correct for historical transactions, as the protocol parameters might differ.

**Suggestion**

Extend the evaluateTx endpoint to optionally accept protocol parameters to avoid querying the state. (e.g. when the limits were continuously increased for exunits). 

Edit: realized that there was already an additionalUtoxs for the evaluate

**Comment by KtorZ** [comment]
That's a good point which sounds quite doable as well. Providing user-define protocol parameters (possibly only a partial subset that overrides existing ones). 

May I ask however, what's the use case you're thinking of? Are you building some kind of validation tool that re-validates historical transactions and re-calculate execution units for these?

**Reply by szist** [reply]
Yup, but it's very low priority for now. Having the additional utxos available seems like the more important thing.
